src/libEth/Eth.cpp

```diff
@@ -145,6 +160,172 @@ EthFields parseRawTxFields(std::string const &message) {
   return ret;
 }
 
+EthFields parseEip2930Transaction(zbytes const& asBytes) {
+  EthFields ret;
+
+  dev::RLP rlpStream1(asBytes,
+                      dev::RLP::FailIfTooBig | dev::RLP::FailIfTooSmall);
+
+  if (rlpStream1.isNull()) {
+    LOG_GENERAL(WARNING, "Failed to parse RLP stream in raw TX!");
+    return {};
+  }
+
+  int i = 0;
+
+  ret.version = DataConversion::Pack(CHAIN_ID, TRANSACTION_VERSION_ETH_EIP_2930);
+
+  // RLP TX contains: chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, signatureYParity, signatureR, signatureS
+  for (const auto& it : rlpStream1) {
+    switch (i) {
+      case 0: // Chain ID - validated earlier
+        break;
+      case 1:
+        ret.nonce = uint32_t(it);
+        break;
+      case 2:
+        ret.gasPrice = uint128_t{it};
+        break;
+      case 3:
+        ret.gasLimit = uint64_t{it};
+        break;
+      case 4:
+        ret.toAddr = zbytes(it);
+        break;
+      case 5:
+        ret.amount = uint128_t(it);
+        break;
+      case 6:
+        ret.code = zbytes(it);
+        break;
+      case 7:
+        ret.accessList = AccessList(it);
+        break;
+      case 8:  // signatureYParity - only needed for pub sig recovery
+        break;
+      case 9:  // signatureR
+      {
+        zbytes b = dev::toBigEndian(dev::u256(it));
+        ret.signature.insert(ret.signature.end(), b.begin(), b.end());
+      } break;
+      case 10:  // signatureS
+      {
+        zbytes b = dev::toBigEndian(dev::u256(it));
+        ret.signature.insert(ret.signature.end(), b.begin(), b.end());
+      } break;
+      default:
+        LOG_GENERAL(WARNING, "too many fields received in rlp!");
+    }
+
+    ++i;
+  }
+
+  // Because of the way Zil handles nonces, we increment the nonce here
+  ret.nonce++;
+
+  return ret;
+}
+
+EthFields parseEip1559Transaction(zbytes const& asBytes) {
+  EthFields ret;
+
+  dev::RLP rlpStream1(asBytes,
+                      dev::RLP::FailIfTooBig | dev::RLP::FailIfTooSmall);
+
+  if (rlpStream1.isNull()) {
+    LOG_GENERAL(WARNING, "Failed to parse RLP stream in raw TX!");
+    return {};
+  }
+
+  int i = 0;
+
+  ret.version = DataConversion::Pack(CHAIN_ID, TRANSACTION_VERSION_ETH_EIP_1559);
+
+  // RLP TX contains: chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list, signature_y_parity, signature_r, signature_s
+  for (const auto& it : rlpStream1) {
+    switch (i) {
+      case 0: // Chain ID - validated earlier
+        break;
+      case 1:
+        ret.nonce = uint32_t(it);
+        break;
+      case 2:
+        ret.maxPriorityFeePerGas = uint128_t{it};
+        break;
+      case 3:
+        ret.maxFeePerGas = uint128_t{it};
+        break;
+      case 4:
+        ret.gasLimit = uint64_t{it};
+        break;
+      case 5:
+        ret.toAddr = zbytes(it);;
```

Comment:
Two ;;
Can you please check & confirm that this will indeed invoke the operator() ? I'm not entirely sure in this case. You can instead try static_cast<zbytes>(it) if it doesn't.

Reply:
Sorry, I don't understand what you mean by "confirm that this will indeed invoke the operator()"? (I'm probably missing some C++ knowledge here)

It is definitely reading the RLP bytes and converting them to a  `zbytes` though, because the tests still parse. If that wasn't happening we would get signature verification failures.

Reply:
The old code invoked the conversion operator() explicitly, so to make it equivalent I suggested you check that this change does the same instead of invoking the constructor and then copying or converting.
If you're saying that this is doing what it's supposed to do then I'd say it's fine.